compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -690,6 +690,7 @@ impl<'tcx> InferCtxt<'tcx> {
             typeck_results: None,
             fallback_has_occurred: false,
             normalize_fn_sig: Box::new(|fn_sig| fn_sig),
+            autoderef_steps: Box::new(|ty| vec![(ty, vec![])]),
```

Comment:
can this ICE instead?

Reply:
Hm, why?

Reply:
my hope was that we only rely on `autoderef_steps` in errors during typeck. I guess we also autoderef when checking method receivers?

If so, don't mind this.

Reply:
So right now the current users of `autoderef_steps` are obligation cause codes that only come from typeck, but there's nothing stopping someone adding a check that uses this during other trait error reporting code, for example, so I'd rather not ICE here.

I'll rename it to `autoderef_steps_for_diagnostic` so people know it's best effort...

Reply:
r=me after adding debug_assert